### ISSUE TITLE
The Multishipping::beforeSetShippingMethods() method can be simplified by removing unused $methods argument

### ISSUE THREAD

**Issue by dmitrii-fediuk** [magento/magento2#3348]
https://mage2.pro/t/490

https://github.com/magento/devdocs/issues/516, https://github.com/magento/magento2/issues/3326
See also: https://github.com/magento/magento2/issues/3340

**Comment by ishakhsuvarov** [comment]
@dfediuk Thank you for reporting this issue. We have created an internal ticket `MAGETWO-49548` to improve the method in question.

**Comment by okobchenko** [comment]
@dfediuk We have investigated reported problem. Unused formal parameter was added for consistency between plugin and pluginized method. There is a convention requiring that "before" plugin receives subject (instance of pluginized object) and list of input parameter of pluginized method. Theoretically particular code can work with $methods argument removed, but this will decrease visibility of relation between methods and arguments exchange and is not recommended practice.
